.github/workflows/ci.yml

```diff
@@ -151,6 +150,7 @@ jobs:
 
     steps:
       - name: Checkout
+        if: needs.changed-files.outputs.run_tests == 'true'
```

Comment:
Just a thought, but instead of adding this `if` to all the steps in `units`, what if `units` sets a variable, and we create a new job that checks the files-changes and the output of `units`?

Reply:
I wanted to avoid having another job in the test matrix and then having to change the repo settings to check the status of that job. The duplicate conditionals are a bit undesirable in the job config, but it makes for a nicer output in the Actions console.